allennlp/service/predictors/coref.py

```diff
@@ -0,0 +1,61 @@
+from overrides import overrides
+
+import spacy
+
+from allennlp.common.util import JsonDict, sanitize
+from allennlp.data import DatasetReader, Instance
+from allennlp.models import Model
+from allennlp.service.predictors.predictor import Predictor
+
+
+@Predictor.register("coreference-resolution")
+class CorefPredictor(Predictor):
+    """
+    Wrapper for the :class:`~allennlp.models.coreference_resolution.CoreferenceResolver` model.
+    """
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+        super().__init__(model, dataset_reader)
+        self._spacy = spacy.load("en", ner=False, tagger=False, vectors=False)
```

Comment:
There's some code in the tokenizer to make sure we only load spacy once.  Can you use that here?

Reply:
Also adding a comment about how you can't just use one of our tokenizers, because you need sentence splitting also, would be good.

Reply:
We do that for the tokenisers because we might load many instances of the class though right? Doesn't seem applicable here - the service only ever loads a single `Predictor` per model, and `predict` can't really cache spacy in memory because it's a command line tool and it only predicts on a single model at a time.

Reply:
I mean this: https://github.com/allenai/allennlp/blob/766911d6397c137248d4dab1f2d77a4c2f74a7be/allennlp/data/tokenizers/word_splitter.py#L181-L193

The issue is that loading spacy is expensive, and it has frequently caused us to run into memory errors on Travis when running tests, if we're not careful about only loading spacy once.  It might be worth putting that block above into a `common.util`, so you can call it here.